libchisel/src/dropsection.rs

```diff
@@ -0,0 +1,116 @@
+use super::{ModuleError, ModuleTranslator};
+
+use parity_wasm::builder::*;
+use parity_wasm::elements::*;
+
+/// Enum on which ModuleTranslator is implemented.
+pub enum DropSection<'a> {
+    NamesSection,
+    /// Name of the custom section.
+    CustomSectionByName(&'a String),
```

Comment:
Perhaps another mode for DropSection would be identifying data sections that are used for rust debugging and panic messages. 
Identifying these is not exactly straightforward, but one option is just using a plain regex lookup for things like `libcore` in the data sections. Not sure what the risk of false positives for this would be.